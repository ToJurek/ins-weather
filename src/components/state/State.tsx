import React from 'react';
import './State.css';
const errorIcon = require('../../images/error.png')

interface Props {
    error: string,
    isLoaded: boolean
}

function renderLoader() {
    return (
        <div className="spinner">
            <div className="bounce1"></div>
            <div className="bounce2"></div>
            <div className="bounce3"></div>
        </div>

    )
}

function renderErrorMessage(error: string) {
    return (
        <div>
            <img src={errorIcon} alt={'error'}
                 className={'error-img'}/>
            <span className={'error-message'}>{error}</span>
        </div>
    )
}

function State(props: Props) {
    return (
        <>
            {!props.isLoaded && !props.error && renderLoader()}
            {props.error && renderErrorMessage(props.error)}
        </>
    )
}

export default State
