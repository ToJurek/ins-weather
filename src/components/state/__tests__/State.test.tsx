import React from "react";
import Enzyme, {mount} from 'enzyme';
import State from "../State";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";


Enzyme.configure({adapter: new Adapter()});

describe('state', function () {
    test('should match to snapshot', () => {
        let wrapper = mount(<State error={''} isLoaded={false}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('should render loader', () => {
        let wrapper = mount(<State error={''} isLoaded={false}/>)
        expect(wrapper.find('.spinner').exists()).toBeTruthy()
    })

    test('should render error message', () => {
        let wrapper = mount(<State error={'Some error'} isLoaded={true}/>)
        expect(wrapper.find('.error-img').exists()).toBeTruthy()
        expect(wrapper.find('.error-message').text()).toBe('Some error')
    })

    test('should render nothings', () => {
        let wrapper = mount(<State error={''} isLoaded={true}/>)
        expect(wrapper.find('.spinner').exists()).toBeFalsy()
        expect(wrapper.find('.error-img').exists()).toBeFalsy()
    })

    test('should render loader', () => {
        let wrapper = mount(<State error={'Some error'} isLoaded={false}/>)
        expect(wrapper.find('.error-img').exists()).toBeTruthy()
        expect(wrapper.find('.error-message').text()).toBe('Some error')
    })
});
