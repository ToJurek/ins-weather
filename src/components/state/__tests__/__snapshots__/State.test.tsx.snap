// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state should match to snapshot 1`] = `
<State
  error=""
  isLoaded={false}
>
  <div
    className="spinner"
  >
    <div
      className="bounce1"
    />
    <div
      className="bounce2"
    />
    <div
      className="bounce3"
    />
  </div>
</State>
`;
